strawberryfields/apps/qchem/dynamics.py

```diff
@@ -250,3 +255,77 @@ def prob(samples: list, excited_state: list) -> float:
         raise ValueError("The excited state must not contain negative values")
 
     return samples.count(excited_state) / len(samples)
+
+
+def sample_coherent(
+    alpha: list, t: float, Ul: np.ndarray, w: np.ndarray, n_samples: int, loss: float = 0.0,
+) -> list:
+    r"""Generate samples for simulating vibrational quantum dynamics with an input coherent state.
+
+    **Example usage:**
+
+    >>> alpha = [0, 1.4]
+    >>> t = 10.0
+    >>> Ul = np.array([[0.707106781, -0.707106781],
+    >>>                [0.707106781, 0.707106781]])
+    >>> w = np.array([3914.92, 3787.59])
+    >>> n_samples = 5
+    >>> sample_coherent(alpha, t, Ul, w, n_samples)
+    [[0, 2], [0, 4], [0, 3], [0, 1], [0, 2]]
+
+    Args:
+        alpha (list): list of displacement parameters
```

Comment:
We should be clear about what we mean by displacement parameters, there are multiple options:
- We only displace in the `x` quadrature, so `alpha` is a list of `x` displacements (what we do now)
- We allow displacement in `p` as well, in which case:
  - is `alpha` a list `[[1, 3], [0.4, 9]]` that is a list of pairs with each pair being the `x` and `p` displacement
  - is `alpha` a list `[[1, np.pi / 3], [2, np.pi / 4]]` a list of pairs with each pair being displacement magnitudes and angles (this is more likely what `DGate` expects
- is `alpha` a list of dimension twice the number of modes where first the `x` displacements are given and then the `p`.

I'm ok with what we have now as a first version unless we have a justification to go more complex, but it would be good to make this clear in the docstring.

Reply:
hmm, I agree. we can extend from just the displacement magnitudes and add angles information too. I will update the function accordingly.